deploy/examples/subvolumegroup.yaml

```diff
@@ -7,3 +7,8 @@ metadata:
 spec:
   # filesystemName is the metadata name of the CephFilesystem CR where the subvolume group will be created
   filesystemName: myfs
+  # reference https://docs.ceph.com/en/quincy/cephfs/fs-volumes/#pinning-subvolumes-and-subvolume-groups
+  # pinning:
+  #   enabled: true   # by fault set to true with default values(type=distributed,setting=1)
+  #   type: distributed
+  #   setting: 1
```

Comment:
To continue the conversation from [here](https://github.com/rook/rook/pull/13283#discussion_r1408267277)...

I'm not sure yet how to interpret these values, but it's boiling down to two approaches for me.

1. Pinning is either enabled or not, and Rook will translate `enabled: true` --> 1 and `enabled: false` --> 0. The default is `enabled: true`.
```yaml
pinning:
  type: distributed | random | export
  enabled: true | false
```
OR

2. The setting should be a string to allow values of "0.0", "1.0", "-1", etc. The default is "1". If the user wants to disable it, they would set "0". But what does it mean to set other values such as "0.5" with random? Or what if they set "0.5" with "distributed"? Does that make sense or will ceph fail for us if we give it a nonsensical combination like that?
```yaml
pinning:
  type: distributed | random | export
  setting: "1"
```
@batrick @parth-gr @Rakshith-R Do one of those make more sense, or what settings are needed?

Reply:
I think we should go with the second flow, with enable parameter(by default true), if someone wants to opt out they can choose `enable:false`

```
pinning:
  type: distributed | random | export
  setting: "1"
  enable: true
```

And with specific pintype we can have checks to the pinsetting values(maybe at CRD or code level).

Reply:
Ok, discussed with @batrick and is more clear now. Based on the various settings for the pin types he mentioned [here](https://github.com/rook/rook/pull/13283#discussion_r1407841829), I'd suggest the schema:
```yaml
pinning:
  type: distributed | random | export
  setting: "1"
```
Where the default is `distributed: 1` and each type has a different range of values:
- distributed: 0 | 1  # 0 is disabled
- export: -1 to 256 # -1 is disabled, and 0-256 are the filesystem ranks
- random: 0.0 to 1.0 # 0.0 is disabled

If they want to disable the pinning, they would just set the corresponding value to disable it.

In the schema, we can only do basic validation that the string is a valid number, unless perhaps CEL rules give us more flexibility based on the type.

Reply:
I see now the CEL rules were added with the type `CephFilesystemSubVolumeGroupSpecPinning`, that looks good instead of using a string, I'm just slow to catch up!

Reply:
@batrick please tell us more about this:

> The default should be distributed=1.

Is this a default that Ceph uses internally? 
If not, please justify why Rook should use this default while Ceph does not. We try to rely on Ceph as the best source of truth to implement its own preferred defaulting behavior as much as possible.

Reply:
> Is this a default that Ceph uses internally?

Ceph provides no default. The impetus for this work was for Rook to set such a default and allow changing it.

> If not, please justify why Rook should use this default while Ceph does not. We try to rely on Ceph as the best source of truth to implement its own preferred defaulting behavior as much as possible.

We did not want to set a default because it could surprise operators who've already configured their cluster and may use the volumes plugin for unknown purposes. However, within the context of Rook and CSI, it always makes sense to set distributed pins by default. Especially since operators have been unable to set these pins except manually outside of Rook.